### PR TITLE
Migrate LB controller to LoadBalancerEndpointSlice

### PR DESCRIPTION
**Pull request by zolug**
Consumer-side migration for #178. Switches the stateless-load-balancer controller from Kubernetes core `EndpointSlice` to the custom `LoadBalancerEndpointSlice` CRD.

Depends on: #194 (DG controller / producer side)

### Summary

The LB controller now reads endpoint data directly from typed fields instead of parsing Zone strings and correlating across per-family slices.

### Key changes

**Production code:**
- `controller.go` — Watch `LoadBalancerEndpointSlice` instead of `discoveryv1.EndpointSlice`, register field indexer for `spec.distributionGroupName`
- `targets.go` — Read `Identifier` field directly (no Zone parsing), `Ready` as bool, dual-stack IPs from single endpoint, first-occurrence-wins for duplicate identifiers, ownerRef verification
- `flows.go` — `hasReadyEndpoints` rewritten with field index lookup + ownerRef check

**Design decisions:**
- **Field indexer + ownerRef verification**: Uses `client.MatchingFields` for efficient lookup, then verifies ownerReference as defense-in-depth against manually-created slices
- **First-occurrence-wins**: During transients, the same identifier may briefly appear in multiple slices; sorted-first slice takes priority (aligns with DG controller's compaction direction)
- **No IP accumulation across slices**: Unlike the old model (IPv4 + IPv6 in separate slices), dual-stack is now in a single endpoint entry — no cross-slice correlation needed

**RBAC:**
- LB Pod Role: replaced `discovery.k8s.io/endpointslices` with `meridio-2.nordix.org/loadbalancerendpointslices`

**Tests:**
- All `reconcileTargets` tests rewritten for new model
- `reconcileFlows` tests updated with field indexer on fake client
- `endpointSliceEnqueue` tests migrated to `LoadBalancerEndpointSlice`
- Removed obsolete Zone-parsing and cross-slice accumulation tests
- Added dual-stack single-endpoint test

### CRD cherry-pick

Includes the `LoadBalancerEndpointSlice` CRD definition (cherry-picked from #194) as a build dependency. The CRD commit is shared between both PRs.

### Testing

- `go build ./...` ✅
- `go test ./internal/controller/loadbalancer/` ✅ (30/30 pass)
- `make lint` ✅
- Manual deployment verified with DG controller producing slices

### CODE REVIEW

File: api/v1alpha1/loadbalancerendpointslice_types.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+// LoadBalancerEndpointSliceSpec defines the desired state of LoadBalancerEndpointSlice.
+type LoadBalancerEndpointSliceSpec struct {
+
+	// distributionGroupName is the name of the DistributionGroup that owns this slice.
+	// Same-namespace relationship is enforced by ownerReference.
+	// +kubebuilder:validation:MinLength=1
+	DistributionGroupName string `json:"distributionGroupName"`
+
+	// gatewayRef identifies the Gateway this slice is scoped to.
+	// Enables per-Gateway endpoint discovery by the LB controller.
```
Comment (ljkiraly) [inline api/v1alpha1/loadbalancerendpointslice_types.go:32]: Minor AI review suggestion:
### One Observation

The `gatewayRef` field is described as "Enables per-Gateway endpoint discovery by the LB controller" but the LB controller does **not** filter by it — it only filters by `distributionGroupName` (field index) and ownerReference. The LB relies on the higher-level `belongsToGateway()` check to ensure it only processes DGs for its Gateway, which transitively ensures it only sees slices relevant to its Gateway. This is correct and sufficient because:
1. Each DistributionGroup is scoped to exactly one Gateway (enforced by the DG controller's single-Gateway restriction)
2. The LB controller only reconciles DGs that pass `belongsToGateway()`
3. Slices owned by a DG are inherently scoped to that DG's Gateway

However, the field comment is slightly misleading — `gatewayRef` serves more as metadata/observability than as a filtering mechanism for the LB. This is fine for v1alpha1 but worth clarifying in comments.

**Review by ljkiraly — APPROVED** [review]
The code is correct and well-tested.

Consider a minor documentation update in this PR or the other one: the `docs/controllers/distributiongroup.md` context entry mentions "EndpointSlice" as the backend type in the architecture table. It should be updated to mention `LoadBalancerEndpointSlice`.

Some future task related to this change, which can be covered in follow-up issues:
1/ Missing test: duplicate identifier across slices: The "first-occurrence-wins" logic is documented but not tested. A test with two slices both containing `identifier=0` would validate the sorting + skip behavior.
2/ **Consider adding `gatewayRef` to the field indexer** — While not needed today (single DG → single Gateway), if future scenarios involve shared DGs across Gateways, the LB could use `client.MatchingFields{"spec.gatewayRef.name": c.GatewayName}` for more efficient filtering.

File: api/v1alpha1/loadbalancerendpointslice_types.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LoadBalancerEndpointSliceSpec defines the desired state of LoadBalancerEndpointSlice.
+type LoadBalancerEndpointSliceSpec struct {
+
+	// distributionGroupName is the name of the DistributionGroup that owns this slice.
+	// Same-namespace relationship is enforced by ownerReference.
+	// +kubebuilder:validation:MinLength=1
```
Comment (ljkiraly) [inline api/v1alpha1/loadbalancerendpointslice_types.go:28]: Be careful when merging: the 'lb-slice' branch has the XValidation:rule="self == oldSelf" immutability validation, added in a later commit.

**Comment by zolug** [comment]
> Consider a minor documentation update in this PR or the other one: the docs/controllers/distributiongroup.md context entry mentions "EndpointSlice" as the backend type in the architecture table. It should be updated to mention LoadBalancerEndpointSlice.

> Be careful when merging: the 'lb-slice' branch has the XValidation:rule="self == oldSelf" immutability validation, added in a later commit.

This PR only contains the LB part except for the CRD changes, and is solely meant for easing the review process.
